packages/samples/headless-ssr/src/app/react/components/search-page.tsx

```diff
@@ -0,0 +1,50 @@
+'use client';
+
+import {
+  SearchSSRState,
+  SearchCSRState,
+  hydrateInitialState,
+  CSRProvider,
+  SSRStateProvider,
+} from '@/src/app/react/common/engine';
+import {useEffect, useState} from 'react';
+import {HydrationMetadata} from '../../common/hydration-metadata';
+import ResultList from './result-list';
+import SearchBox from './search-box';
+
+export default function SearchPage({ssrState}: {ssrState: SearchSSRState}) {
+  const [csrResult, setCSRResult] = useState<SearchCSRState | undefined>(
+    undefined
+  );
+
+  useEffect(() => {
+    hydrateInitialState(ssrState).then(({engine, controllers}) => {
+      setCSRResult({engine, controllers});
+    });
+  }, [ssrState]);
+
+  const content = (
+    <>
+      <SearchBox />
+      <ResultList />
+      <HydrationMetadata ssrState={ssrState} csrResult={csrResult} />
+    </>
+  );
```

Comment:
Unlike the other sample, I think we should take all these these components (except `HydrationMetadata`) as `children`, since we don't need to manually give them the state as parameters anymore. It was part of my motivation behind generating providers.

The `SearchPage` component here could be renamed `SearchPageProvider` or something like that.

Reply:
Done.